src/screens/FeedScreen/hooks/useCoreFeed.js

```diff
@@ -222,28 +227,7 @@ const useCoreFeed = () => {
   };
 
   const sendViewPostTime = async (withResetTime = false) => {
-    const currentTime = new Date();
-    const diffTime = currentTime.getTime() - timer.getTime();
-    const id = feeds?.[viewPostTimeIndex]?.id;
-    if (!id) return;
-
-    viewTimePost(id, diffTime, SOURCE_FEED_TAB);
-    if (withResetTime) setTimer(new Date(), dispatch);
-  };
-
-  const getCurrentPostViewed = (momentumEvent) => {
-    const {y} = momentumEvent.nativeEvent.contentOffset;
-    const shownIndex = Math.ceil(y / dimen.size.FEED_CURRENT_ITEM_HEIGHT);
-    return shownIndex;
-  };
-
-  const updateViewPostTime = (momentumEvent) => {
-    setViewPostTimeIndex(getCurrentPostViewed(momentumEvent), dispatch);
-    setTimer(new Date(), dispatch);
-  };
-
-  const isSamePostViewed = (momentumEvent) => {
-    return getCurrentPostViewed(momentumEvent) === viewPostTimeIndex;
+    sendViewPostTimeWithFeeds(feeds, withResetTime);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `sendViewPostTimeWithFeeds` is being called with the correct parameters. However, make sure that the `feeds` state variable is up-to-date when this function is called. If there are any asynchronous operations that update `feeds`, they should be awaited before calling this function to avoid sending stale data.



<!-- This is an auto-generated comment by OSS CodeRabbit -->